sql/CMakeLists.txt

```diff
@@ -34,47 +34,110 @@ set(EXT_SQL_EXTRA_FILES
+SET(EXT_SQL_SPECIAL_FILES
+    timescaledb--0.6.1--0.7.0.sql
+    timescaledb--0.7.0--0.7.1.sql
+  )
+
+set(EXT_SQL_UPDATE_PRE_GLOBAL_FILE updates/pre-global.sql)
+set(EXT_SQL_UPDATE_POST_GLOBAL_FILE updates/post-global.sql)
+set(UPDATE_FILE_LIST ${EXT_SQL_UPDATE_PRE_GLOBAL_FILE} ${EXT_SQL_UPDATE_PRE_FILE} ${SQL_FILES} ${EXT_SQL_UPDATE_POST_FILE}  ${EXT_SQL_UPDATE_POST_GLOBAL_FILE})
+
+#replace MODULE PATHNAME for files making up EXT_SQL_FILE (timescaledb--version.sql) and EXT_SQL_UPDATE_FILE (timescaledb--previous_version--version.sql)
+set(MODULE_PATHNAME "$libdir/timescaledb-${PROJECT_VERSION_MOD}")
+set(UPDATE_FILE_LIST_VERSIONED "")
+set(SQL_FILES_VERSIONED "")
+FOREACH(update_file ${UPDATE_FILE_LIST})
+  set(update_file_versioned ${update_file}.${PROJECT_VERSION_MOD})
+  configure_file(${update_file} ${update_file_versioned} @ONLY)
+  list(APPEND UPDATE_FILE_LIST_VERSIONED ${CMAKE_CURRENT_BINARY_DIR}/${update_file_versioned})
+  if (${update_file} IN_LIST SQL_FILES)
+    list(APPEND SQL_FILES_VERSIONED ${CMAKE_CURRENT_BINARY_DIR}/${update_file_versioned})
+  endif()
+ENDFOREACH(update_file)
+
+function(cat_files SRC_FILE_LIST OUTPUT_FILE)
+  if (WIN32)
+    # Make list of files into string of files separated by "+"
+    # to make Windows copy concatenate them
+
+    file(TO_NATIVE_PATH "${SRC_FILE_LIST}" SRC_FILE_LIST_NATIVE)
+    string(REPLACE ";" ";+" SQL_LIST_JOINED "${SRC_FILE_LIST_NATIVE}")
+
+    file(TO_NATIVE_PATH "${OUTPUT_FILE}" OUTPUT_FILE_NATIVE)
+
+    set(CAT_CMD copy /B /y ${SQL_LIST_JOINED} "\"${OUTPUT_FILE_NATIVE}\"" >NUL)
+  else ()
+    set(CAT_CMD cat ${SRC_FILE_LIST} > ${OUTPUT_FILE})
+  endif ()
+  add_custom_command(
+    OUTPUT ${OUTPUT_FILE}
+    DEPENDS ${SRC_FILE_LIST}
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    COMMAND ${CAT_CMD}
+    COMMENT "Generating ${OUTPUT_FILE}"
+  )
+endfunction()
+
+cat_files("${SQL_FILES_VERSIONED}" ${CMAKE_CURRENT_BINARY_DIR}/${EXT_SQL_FILE})
 add_custom_target(sqlfile ALL DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/${EXT_SQL_FILE})
 
-# Command and target for the update SQL file
-add_custom_command(
-  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${EXT_SQL_UPDATE_FILE}
-  DEPENDS ${SQL_FILES} ${EXT_SQL_UPDATE_PRE_FILE} ${EXT_SQL_UPDATE_POST_FILE}
-  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-  COMMAND ${CAT_SQL_UPDATE_FILE_CMD}
-  COMMENT "Generating ${EXT_SQL_UPDATE_FILE}")
+cat_files("${UPDATE_FILE_LIST_VERSIONED}" ${CMAKE_CURRENT_BINARY_DIR}/${EXT_SQL_UPDATE_FILE})
+add_custom_target(sqlupdatefile ALL DEPENDS  ${CMAKE_CURRENT_BINARY_DIR}/${EXT_SQL_UPDATE_FILE})
+
+#This generates the update file without MODULE_PATHNAME replaced. This file should be copied for the release.
+cat_files("${UPDATE_FILE_LIST}" ${CMAKE_CURRENT_BINARY_DIR}/${EXT_SQL_UPDATE_FILE}.release)
+add_custom_target(sqlreleaseupdate ALL DEPENDS  ${CMAKE_CURRENT_BINARY_DIR}/${EXT_SQL_UPDATE_FILE}.release)
+
+#You want to create timescaledb-prev_version--current_version for all possible prev_versions.
```

Comment:
This logic is not tested. I have no idea how to test CMAKE scripts. @erimatnor any ideas?